### PR TITLE
Add functions to yesod-test to support the new CSRF middleware

### PR DESCRIPTION
**Pull request by MaxGabriel**
This PR adds functions to add CSRF tokens from cookies, supporting the new CSRF middleware added in #1017.

### CODE REVIEW

**Comment by MaxGabriel** [comment]
@snoyberg or @gregwebs Can you upload yesod-core-1.4.14 to Hackage so that yesod-test can depend on it?

File: yesod-test/Yesod/Test.hs [hunk 1]
```diff
@@ -93,6 +95,7 @@ module Yesod.Test
     -- * Grab information
     , getTestYesod
     , getResponse
+    , getRequestCookies
```
Comment (MaxGabriel) [inline yesod-test/Yesod/Test.hs]: I don't know the best spot for this function, either in the export list or in the source code.

**Comment by snoyberg** [comment]
Uploaded

File: yesod-test/Yesod/Test.hs (hunk 1 above)
Comment (gregwebs) [inline yesod-test/Yesod/Test.hs]: looks fine to me

**Comment by gregwebs** [comment]
Thanks for handling this issue from every angle :) Testing is often forgotten about

**Pull request merged**